server/src/main/java/org/cloudfoundry/identity/uaa/util/TokenValidation.java

```diff
@@ -133,8 +136,32 @@ public TokenValidation checkIssuer(String issuer) {
             throw new InvalidTokenException("Token does not bear an ISS claim.", null);
         }
 
-        if (!equals(issuer, claims.get(ISS))) {
-            throw new InvalidTokenException("Invalid issuer (" + claims.get(ISS) + ") for token did not match expected: " + issuer, null);
+        if (validationMode == XOAuthIssuerValidationMode.STRICT) {
```

Comment:
I think this would be better as a `switch` with the default case set to throw an error due to an unknown enum value.